flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/sql/ml/SqlMLEvaluateTableFunction.java

```diff
@@ -0,0 +1,208 @@
+    @Override
+    public boolean argumentMustBeScalar(int ordinal) {
+        return ordinal != 0;
+    }
+
+    @Override
+    protected RelDataType inferRowType(SqlOperatorBinding opBinding) {
+        final RelDataTypeFactory typeFactory = opBinding.getTypeFactory();
+        final RelDataType inputRowType = opBinding.getOperandType(0);
+
+        // Create a MAP type for evaluation results
+        RelDataType keyType = typeFactory.createSqlType(SqlTypeName.VARCHAR);
+        RelDataType valueType = typeFactory.createSqlType(SqlTypeName.DOUBLE);
+        RelDataType mapType = typeFactory.createMapType(keyType, valueType);
+
+        return typeFactory
+                .builder()
+                .kind(inputRowType.getStructKind())
+                .add("result", mapType)
+                .build();
+    }
+
+    private static class EvaluateOperandMetadata implements SqlOperandMetadata {
+        private static final List<String> PARAM_NAMES =
+                List.of(
+                        PARAM_INPUT,
+                        PARAM_MODEL,
+                        PARAM_LABEL,
+                        PARAM_FEATURE,
+                        PARAM_TASK,
+                        PARAM_CONFIG);
+        private static final List<String> MANDATORY_PARAM_NAMES =
+                List.of(PARAM_INPUT, PARAM_MODEL, PARAM_LABEL, PARAM_FEATURE);
+
+        EvaluateOperandMetadata() {}
+
+        @Override
+        public List<RelDataType> paramTypes(RelDataTypeFactory typeFactory) {
+            return Collections.nCopies(
+                    PARAM_NAMES.size(), typeFactory.createSqlType(SqlTypeName.ANY));
+        }
+
+        @Override
+        public List<String> paramNames() {
+            return PARAM_NAMES;
+        }
+
+        @Override
+        public boolean checkOperandTypes(SqlCallBinding callBinding, boolean throwOnFailure) {
+            if (!SqlValidatorUtils.checkTableAndDescriptorOperands(callBinding, 2, 3)) {
+                return SqlValidatorUtils.throwValidationSignatureErrorOrReturnFalse(
+                        callBinding, throwOnFailure);
+            }
+
+            if (!SqlValidatorUtils.throwExceptionOrReturnFalse(
+                    checkModelSignature(callBinding, 3, 2), throwOnFailure)) {
+                return false;
+            }
+
+            if (callBinding.getOperandCount() == PARAM_NAMES.size()) {
+                // Check task and config parameters
+                if (!SqlValidatorUtils.throwExceptionOrReturnFalse(
+                        checkTask(callBinding.operand(4)), throwOnFailure)) {
+                    return false;
+                }
+
+                if (!SqlValidatorUtils.throwExceptionOrReturnFalse(
+                        checkConfig(callBinding.operand(5)), throwOnFailure)) {
+                    return false;
+                }
+            }
+
+            if (callBinding.getOperandCount() == MANDATORY_PARAM_NAMES.size() + 1) {
+                // Last param can be config or task
```

Comment:
I am confused about the comment. Considering the current function signature,
```
return opName
                    + "(TABLE table_name, MODEL model_name, DESCRIPTOR(label_column), DESCRIPTOR(feature_columns), [task], [MAP[]])";
```
The fifth parameter is always task. If not, we should throw a exception to notify users?

Reply:
5th param is optional, it can be task or config map. 6th param is also optional and it must be config if exist

Reply:
The current behaviour is similar that we have a function
```
func([int], [long])
```
When users only specify `func(1)` and framework uses `1` as the second parameter. If users doesn't use named arguments, we should only use the location to map the parameter and value.

Reply:
In SQL, we use BIGINT to parse the literal value if the value is a number without float.